packages/world/ts/internal/genObjectTypeSol.ts

```diff
@@ -276,6 +275,24 @@ ${allCategoryMetadata
     return false;
   }
 
+  function isTillable(ObjectType self) internal pure returns (bool) {
+    ${objects
+      .filter((obj) => obj.isTillable)
+      .map((obj) => `if (self == ObjectTypes.${obj.name}) return true;`)
+      .join("\n    ")}
+    return false;
+  }
+
+  function isPlantableOn(ObjectType self, ObjectType on) internal pure returns (bool) {
```

Comment:
I thought of adding this per object but currently its actually per category, so didnt add it there